{WIP} Add appsody prerequisite

sghung opened the pull request:
PR being tested and reviewed, I will remove {WIP} once I have the writeup and testing listed. The PR is for getting these files on windows for testing.

sghung commented:
One thing I couldn't figure out, is the build. When I check the build generated by grandelw, it does not have the appsody binaries in there. Yet, I see the entire resources folder is part of the build.properties, so I'm not sure if I'm doing something wrong.

makandre commented on dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/InstallUtil.java:
```diff
@@ -309,6 +328,55 @@ public static String getInstallerExecutable() throws IOException {
 		}
 	}
 	
+	protected static void copyExecutable(Map<OperatingSystem, String> fileMap) throws IOException {
```
Is this method largely a copy `getInstallerExecutable`?

I wonder if it might be less duplicate code if we made `installMap` a map of string arrays, then we can just do the copy in a loop.

sghung replied:
I had wanted to create another method that could be reusable, but there are a few things that are different. First the get installer executable will assign the installExec string to it and the log messages are also pretty specific to it. The file check at the start for getInstallerExecutable is also not done. Originally, I really did want to refactor everything to make it generic, however, there is a higher chance to introduce a new regression and this code will not be used once we switch to downloading the binary instead of including it

makandre replied:
Could we at least refactor it so the actual part doing the copy (i.e. the try block is in a common private method - minus the `installExec` assignment which can be moved to the outside)

sghung commented:
> I added a README in [eclipse/codewind-vscode#176](https://github.com/eclipse/codewind-vscode/pull/176) to record the appsody version we bundle

Good idea, I have added the equivalent readme.md

sghung commented:
Although locally, I cannot get the zip to include the new binaries, I see in the jenkins build, it does include the new appsody binaries: https://download.eclipse.org/codewind/codewind-eclipse/pr/266/2/

eharris369 commented on dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/InstallUtil.java:
```diff
@@ -309,6 +328,55 @@ public static String getInstallerExecutable() throws IOException {
+		String execPath = installerDir + File.separator + execName;
+		
+		// Check if file exists, if it does return the path and skip the copying
+		if ((new File(execPath)).exists()) {
```
You might also want to have a variable that indicates the file has been copied over.  If the user is using an old Eclipse workspace with a newer version of Codewind the file might exist in their workspace at an older level.

sghung commented:
Thank you for the feedback, I will refactor the code and update the PR

sghung commented:
@makandre and @eharris369 , please review again. I have refactored the install to make it generic. I created a new class to keep track of the necessary properties and added the appropriate getter and setter methods to keep the existing behaviour.

makandre commented on dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/InstallUtil.java:
```diff
@@ -260,16 +278,25 @@ public static Process runInstaller(String cmd, String... options) throws IOExcep
-		String relPath = installMap.get(os);
+		
+		Map<OperatingSystem, String> osPathMap = operation.getOSPathMap();
+		if (osPathMap == null) {
```
This would never be `null` right?  Since we control how the operations are instantiated, it's all private.
